python/create_init.py

```diff
@@ -26,31 +26,26 @@
 
 src = '''__all__ = ('language', 'stemmer')
 
-%(imports)s
-
-_languages = {
-%(languages)s
-}
-
 try:
     import Stemmer
-    cext_available = True
+    algorithms = Stemmer.algorithms
+    stemmer = Stemmer.Stemmer
 except ImportError:
-    cext_available = False
+%(imports)s
+
+    _languages = {
+%(languages)s
+    }
 
-def algorithms():
-    if cext_available:
-        return Stemmer.algorithms()
-    else:
+    def algorithms():
         return list(_languages.keys())
 
-def stemmer(lang):
-    if cext_available:
-        return Stemmer.Stemmer(lang)
-    if lang.lower() in _languages:
-        return _languages[lang.lower()]()
-    else:
-        raise KeyError("Stemming algorithm '%%s' not found" %% lang)
+    def stemmer(lang):
+        lang = lang.lower()
+        if lang in _languages:
+            return _languages[lang]()
+        else:
+            raise KeyError("Stemming algorithm '%%s' not found" %% lang)
```

Comment:
An alternate approach here could be using `importlib.import_module()` to only load the language requested. `languages` becomes `dict[str, str]` mapping the language name to the stemmer class, and then here you do

```python
try:
    mod = importlib.import_module(f'snowballstemmer.{lang}_stemmer')
    stemmer_class = getattr(mod, _languages[lang])
    return stemmer_class()
except (AttributeError, ImportError, KeyError):
    raise KeyError(f'Stemming algorithm {lang!r} not found') from None
```

Reply:
We could even get rid of `_languages` entirely if each stemmer module provided a common way to create a stemmer of its class - e.g. if I append `factory = EnglishStemmer` to `english_stemmer.py` and similarly for each subclass then I can do:

```
>>> import importlib
>>> mod = importlib.import_module('snowballstemmer.english_stemmer')
>>> stemmer = mod.factory()
>>> stemmer
<snowballstemmer.english_stemmer.EnglishStemmer object at 0x7f1f34b92270>
>>> stemmer.stemWord('stemming')
'stem'
```

There's probably a more pythonic way to do that but hopefully you get the idea.